rust/driver-ipc/src/sync/client.rs

```diff
@@ -35,25 +37,32 @@ impl Client {
         RUNTIME.block_on(self.0.remove_all())
     }
 
-    /// Receive generic reply
-    ///
-    /// If `last` is false, will only receive new messages from the point of calling
-    /// If `last` is true, will receive the the last message received, or if none, blocks until the next one
-    ///
-    /// The reason for the `last` flag is that replies are auto buffered in the background, so if you send a
-    /// request, the reply may be missed
-    pub fn receive_reply(&mut self, last: bool) -> Option<ReplyCommand> {
-        RUNTIME.block_on(self.0.receive_reply(last))
-    }
-
     /// Receive an event. Only new events after calling this are received
     pub fn receive_event(&mut self) -> EventCommand {
-        RUNTIME.block_on(self.0.receive_event())
+        RUNTIME.block_on(async {
+            self.0
+                .receive_events()
+                .next()
+                .await
+                .expect("Stream should never finish")
+        })
+    }
+
+    pub fn add_event_receiver(&mut self, cb: impl Fn(EventCommand) -> bool + Send + 'static) {
+        let mut stream = self.0.receive_events();
+
+        RUNTIME.spawn(async move {
+            while let Some(event) = stream.next().await {
```

Comment:
The changes I mentioned above about doing Result for lagged, should fix this from prematurely terminating because of a lag. (As well as dropping the sender, so it doesn't continue waiting if read prematurely exited)

Reply:
As mentioned, the stream does not close because of a `Lagged` error, so this should be no problem.